cmd/botkube-agent/main.go

```diff
@@ -192,8 +192,8 @@ func run(ctx context.Context) (err error) {
 	defer pluginManager.Shutdown()
 
 	// Health endpoint
-	healthChecker := healthChecker{applicationStarted: false}
-	healthSrv := newHealthServer(logger.WithField(componentLogFieldKey, "Health server"), conf.Settings.HealthPort, &healthChecker)
+	healthChecker := health.NewChecker(ctx, conf, pluginHealthStats)
+	healthSrv := healthChecker.NewServer(logger.WithField(componentLogFieldKey, "Health server"), conf.Settings.HealthPort)
```

Comment:
let's move starting the health server to line 189:
```go
	// Health endpoint
	healthChecker := health.NewChecker(ctx, conf, pluginHealthStats)
	healthSrv := healthChecker.NewServer(logger.WithField(componentLogFieldKey, "Health server"), conf.Settings.HealthPort)
	errGroup.Go(func() error {
		defer analytics.ReportPanicIfOccurs(logger, reporter)
		return healthSrv.Serve(ctx)
	})
```

this way it will be available before the `	err = pluginManager.Start(ctx)` which may take some time as it downloads all enabled plugins.